cmd/testrunner/main.go

```diff
@@ -215,10 +225,29 @@ func withTestImage(image string) func(obj runtime.Object) {
 	}
 }
 
+func updateEnvVarList(envVarList []corev1.EnvVar, key, val string) []corev1.EnvVar {
```

Comment:
I have a feeling this whole function can be simpler?

Instead of using `indexToReplace`, could we just do something like this?

```golang
for index, envVar := range envVarList {
  if envVar.Name == key {
          envVarList[index] = corev1.EnvVar{Name: key, Value: val}
	  return envVarList
  }
}
return append(envVarList, corev1.EnvVar{Name: key, Value: val})
```

Reply:
That looks good! Done.